docker/init.sh

```diff
@@ -0,0 +1,48 @@
+#! /bin/bash
+
+set -e
+
+# data directory is required for caching the secret key in a file
+if [ "$DATA_DIR" == "" ]; then
+    echo -n "Error: DATA_DIR environment variable not set!"
+    echo "Are you sure you are running this script in a Docker container?"
+    exit 1
+fi
+
+SECRET_KEY_FILE="$DATA_DIR/secret_key"
+
+if [ ! -f "$SECRET_KEY_FILE" ]; then
+    echo ">>> Creating a new secret key file..."
+    install -m 0600 /dev/null "$SECRET_KEY_FILE"
+    SECRET_KEY=$(bundle exec rails secret)
```

Comment:
As you should can see, the file already _uses_ the `bundle` command to generate the secret. But the secret should be cached across multiple containers. In Docker world, containers are disposable things, they should not contain any permanent data, so one has to save it to volumes. Please see the Docker docs for more information. So, just calling the command is not enough.